### ISSUE TITLE
[NIGHTWACH E2E]: Public Websites - Progressbar must be updated for axe violation in Site / Tests / Home / 00 Required E2e Spec

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#21799]
# [508-defect-1 :exclamation: Launchblocker](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix issue.
1. At message the accessibility specialist when issue is resolved.

## Point of Contact

**VFS Point of Contact:** _Trevor_

## User Story or Problem Statement
Accessibility specialists discovered Nightwatch axe checks failing silently during a recent staging review. Multiple axe violations were discovered where ARIA progressbar attributes were not conforming to valid values. Preliminary research indicates this was for a custom version of the [Design System loading indicator](https://design.va.gov/storybook/?path=/story/components-loadingindicator--default).

## Related Issue:
* [[AUTOMATION]: Update Nightwatch axeCheckJS to capture failing tests ](https://github.com/department-of-veterans-affairs/va.gov-team/issues/21738)

## Acceptance Criteria
- [ ] All instances of the error are fixed
- [ ] Axe-browser plugin or axe-core CLI scans show 0 Section 508, WCAG2A or WCAG2AA violations
- [ ] Nightwatch e2e test `00 Required E2e Spec` passes axeCheck validation OR
- [ ] Test is converted to Cypress and passes axeCheck validation

## Environment
* src/site/tests/home/00-required.e2e.spec.js

**Comment by brianalloyd** [comment]
@ncksllvn I'm going to move this into this sprint based on the fact its `launch blocking` could you scrub into this once you have some bandwidth?  Please reach out with any questions.  Here's the [slack convo ](https://dsva.slack.com/archives/C52CL1PKQ/p1616437198156500) thanks.

**Comment by ncksllvn** [comment]
Sure @brianalloyd no problem

**Comment by 1Copenut** [comment]
@brianalloyd for context, we assign `defect-severity-1` aka launch blocking on automated issues like these for their potential to break builds and ATO language. If there's no issue or it's a different page, feel free to adjust as needed.

**Comment by 1Copenut** [comment]
@ncksllvn I had a thought about this one. If the page is using the Loading Indicator or a version of said, that's probably what's throwing the error. Our current implementation uses a `role="progressbar"` which requires an accessible label with current version of the axe rulesets.

**Comment by brianalloyd** [comment]
@1Copenut that explanation helps as our team was stumped on diagnosing this issue/solution.

**Comment by 1Copenut** [comment]
Glad to hear that @brianalloyd. I'd been puzzling over that for a long while and the thought randomly crossed my mind.

**Comment by ncksllvn** [comment]
I’ll definitely dig into this, but right now I just can’t think of where the homepage uses a loading indicator at all, since it’s a static webpage and not a react app. I’m sure it’ll become clear once I start trying to recreate.

**Comment by ncksllvn** [comment]
Still investigating, but I think the progress bar being described here is actually the Chromium loading indicator when the browser is tarting up. I think the integration test needs to wait for the page to load before issuing its accessibility assertions.

**Comment by ncksllvn** [comment]
@1Copenut Could you describe how I can recreate it this? I can't seem to recreate the failure locally (the e2e test passes w/o error) and I don't see the output in CI.

**Comment by 1Copenut** [comment]
@ncksllvn Hm, the best option I could offer r/n is to checkout my feature branch https://github.com/department-of-veterans-affairs/vets-website/tree/feature/nightwatch-axeCheck and run the Nightwatch accessibility tests again on your local machine.

With the tests failing silently, I opted to rewrite the Nightwatch `axeCheck` helper to test all the major rulesets and allow developers to pass in a custom config object to override individual rules. Here's my WIP PR if you're curious what I'm working on: https://github.com/department-of-veterans-affairs/vets-website/pull/16424/files

**Comment by ncksllvn** [comment]
1. I pulled your branch `feature/nightwatch-axeCheck`
2. I started the website via `yarn watch --env.entry=static-pages`
3. I ran `BUILDTYPE=vagovdev yarn test:e2e -- src/site/tests/home/00-required.e2e.spec.js`
4. I observed the output below -

![image](https://user-images.githubusercontent.com/1915775/112887387-216f2a80-90a1-11eb-9830-4b17299a39a1.png)

I am thinking that somehow the Chromium loading indicator was being tested by the axe check instead of the actual homepage. I can't recreate this and there is no progress bar on the homepage as far as I can tell.

**Comment by 1Copenut** [comment]
Thanks for taking a closer look @ncksllvn. I'm fresh out of ideas. I'll keep an eye on it when I push my next WIP commit. I appreciate you taking extra time to troubleshoot.

**Comment by ncksllvn** [comment]
Thanks for your help @1Copenut!

**Comment by ncksllvn** [comment]
@brianalloyd I think we can mark this as blocked, until we are instructed on how to recreate. It passes in CI and on my computer.

**Comment by brianalloyd** [comment]
@ncksllvn sounds good, thanks for working the details, shifting this into blocked until we can recreate.  Appreciate it.

**Comment by brianalloyd** [comment]
Can you confirm these are still issues, might have had duplicate issues/tickets that we worked. @1Copenut TIA

**Comment by 1Copenut** [comment]
@brianalloyd I'm not seeing this error appear in the Nightwatch e2e or static content tests anymore. I think we've got it handled, so I'll close the issue and keep an eye out. Thank you to you and your team for prioritizing these and getting fixes out so quickly.

/cc @ncksllvn & @kelsonic
